### PR TITLE
 Add null provider to support the privacy API

### PR DESCRIPTION
**Pull request by SammyIsConfused**
Moodle now adheres to the GDPR which means we need to implement a privacy provider. 
See https://docs.moodle.org/dev/Privacy_API#Plugins_which_do_not_store_personal_data for details.  

image

### CODE REVIEW

File: classes/privacy/provider.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+class provider implements \core_privacy\local\metadata\null_provider {
+    //this plugin does not store any personal user data.
+    /**
+     * Get the language string identifier with the component's language
+     * file to explain why this plugin stores no data.
+     */
+    public static function get_reason(): string {
+        return 'privacy:metadata';
+    }
+}
```
Comment (ChristianMurphy) [inline classes/privacy/provider.php:13]:
```suggestion
}

```

File: classes/privacy/provider.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+namespace block_equella_search\privacy;
+
+class provider implements \core_privacy\local\metadata\null_provider {
+    //this plugin does not store any personal user data.
```
Comment (ChristianMurphy) [inline classes/privacy/provider.php:5]:
```suggestion
    // this plugin does not store any personal user data.
```

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Seems fine, just need to make sure we close those php tags. (You may even want to consider doing the same for the other files in this repo.)

Also, same query about the text: https://github.com/openequella/moodle-block_openEQUELLA_links/pull/7#discussion_r477014178

**Comment by ChristianMurphy** [comment]
> just need to make sure we close those php tags

PSR-2, the PHP code style, recommends never adding `?>` to pure PHP files.
https://www.php-fig.org/psr/psr-2/#22-files

**Comment by edalex-ian** [comment]
No probs. https://github.com/openequella/moodle-block_openEQUELLA_tasks/pull/5#discussion_r477034544

**Review by edalex-ian — APPROVED** [review]

**Pull request merged**